sites/main-site/src/content/blog/2024/seqera-containers-part-2.mdx

```diff
@@ -0,0 +1,646 @@
+steps are fully automated.
+
+## Container image creation
+
+As with most automation in nf-core, container creation will happen in GitHub Actions.
+Edits to a module's `environment.yml` file will trigger a workflow that uses the
+[`wave-cli`](https://github.com/seqeralabs/wave-cli) to build the container images.
+
+1. GitHub Actions identifies changes in the `environment.yml` file.
+2. `wave-cli` is executed on the updated environment file.
+3. Seqera Containers builds new containers for various platforms and architectures.
+4. GitHub Actions runs stub tests commits the updated the
+   [version snapshot](https://github.com/nf-core/modules/blob/1fe2e6de89778971df83632f16f388cf845836a9/modules/nf-core/bowtie/align/tests/main.nf.test.snap#L32-L46).
+
+<div class="excalidraw">
+    <Fragment set:html={creation_flow} alt="Container creation flow" />
+</div>
+
+Once this GitHub Actions run completes it will push the new commits back to the PR
+and the regular nf-test CI will run.
+
+## nf-test versions snapshot
+
+One of the primary reasons that we were so excited to adopt nf-test was the snapshot functionality.
+Every test has a snapshot file with the expected outputs, and the outputs are deterministic
+(not a binary file, and there's no dates).
+
+In that snapshot, we also capture the versions of the dependencies for the module
+([example shown for bowtie2](https://github.com/nf-core/modules/blob/1fe2e6de89778971df83632f16f388cf845836a9/modules/nf-core/bowtie/align/tests/main.nf.test.snap#L32-L46)):
+
+```json title="main.nf.test.snap" {4-7}
+    "versions": {
+        "content": [
+            {
+                "BOWTIE_ALIGN": {
+                    "bowtie": "1.3.0",
+                    "samtools": "1.16.1"
+                }
+            }
+        ],
+        "meta": {
+            "nf-test": "0.9.0",
+            "nextflow": "24.04.4"
+        },
+        "timestamp": "2024-09-27T10:42:58.892298"
+    },
+```
+
+This gives a second level of confirmation that the containers were correctly generated.
+
+When updating containers, the nf-test snapshot is parsed and compared to the snapshot
+from before the new containers were built.
+The snapshot changes are discarded if anything other than the `versions` key changed,
+so it should only vary in the software versions reported.
+If any other changes are detected, the snapshot change will be rejected and _not_ committed back to the PR.
+Then PR reviewers will see failing tests and need to manually update the snapshot file.
+
+This means that we can automatically commit the updated snapshot file in the PR if
+the tool output is unchanged, saving the developer from taking this extra step.
+
+:::tip
+
+Note that in the above example, the versions are in the snapshot in plain text,
+not using an md5 hash. This is a change that we will roll out for all modules,
+as it makes verification in the PR much easier.
+
+:::
+
+## Automatic version bumps with Renovate
```

Comment:
This is honestly scaring the shit out of me, I'm expecting huge waves of PRs that will be ignored and never addressed and @SPPearce will rage quit.

The vast majority of 'CODEOWNERS' will never come back to nf-core so will go stale

Reply:
But we can see what happens...

Reply:
Hopefully 90% of them should auto-merge, resulting in the opposite: nf-core not going stale.

A good success story for this is conda-forge, which automatically updates recipes when new PyPI packages comes out - self reviews and self merges.

If it's problematic, it's easy enough to switch off..

Reply:
same